ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/SlidingWindowRateLimiter.java

```diff
@@ -58,13 +64,20 @@
    * Window size over which the throttle is supposed to be enforced measured in milliseconds.
    */
   private final long slidingWindowSizeMs;
+  private final Sensor rejectSensor;
 
   /**
    * Aggregate of query response sizes in the past hour
    */
-  private long numBytesInWindow;
+  private volatile long numBytesInWindow;
```

Comment:
Actually, I think it wasn't even threadsafe before because this object is shared between threads.  I don't recall what are the atomicity guarantees of volatile long, but it might make sense to use an AtomicLong.

Reply:
Java guarantees that volatile longs are atomic, so we should be good there. Thanks!